### ISSUE TITLE
Invalid Unicode escapes in bytes literals

### ISSUE THREAD

**Issue by InSyncWithFoo** [microsoft/pyright#9476]
Pyright seems to be accepting these Unicode escapes; it resolves all of them to `b'\xFF'` regardless of the actual values ([playground](https://pyright-play.net/?pyrightVersion=1.1.389&pythonVersion=3.14&strict=true&code=IYAgvCBGDkA6CuAGRAxF0SZAbhAJwFMA3A4AGwH0AXATwAcCAKYASkwGIQAZASyoLzkA2pABEsAB4AzKaIC6AWABQkcFDjw0aDFlyES5avSaQ2ITr36CyI8dNmKlAYzUxYAVWTIt6PcVKUtAyMTmYWfALCYpIy8soAJq5wnqioWtB%2BBoHGjPFh3BHWtjEOykA)):

```python
a = b'\u00FF'      # Literal[b"\xff"]
b = b'\uFFFF'      # Literal[b"\xff"]
c = b'\U0000FFFF'  # Literal[b"\xff"]
d = b'\U00F0FFFF'  # Literal[b"\xff"]
```

However, `\u` and `\U` escapes are [not recognized](https://docs.python.org/3/reference/lexical_analysis.html#escape-sequences) within bytes literals:

```pycon
Python 3.14.0a1 (tags/v3.14.0a1:8cdaca8, Oct 15 2024, 20:08:21) [MSC v.1941 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> b'\u00FF'
<python-input-0>:1: SyntaxWarning: invalid escape sequence '\u'
  b'\u00FF'
b'\\u00FF'
```

**Comment by erictraut** [comment]
Yeah, pyright's logic doesn't currently check for this syntax error. I think it's a reasonable enhancement request.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.390.
